src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -294,603 +363,1316 @@ public static boolean hasToolLevel(final String tool, final ItemStack stack, fin
     }
 
     /**
-     * Adapted from {@link net.minecraft.entity.player.InventoryPlayer#addItemStackToInventory(ItemStack)}.
+     * Returns the amount of occurrences in the {@link ICapabilityProvider}.
      *
-     * @param inventory Inventory to add itemstack to.
-     * @param itemStack ItemStack to add.
-     * @param building  the building.
-     * @return itemStack which has been replaced.
+     * @param provider                    {@link ICapabilityProvider} to scan.
+     * @param itemStackSelectionPredicate The predicate used to select the
+     *                                    stacks to count.
+     * @return Amount of occurrences of stacks that match the given predicate.
      */
-    @Nullable
-    public static ItemStack forceItemStackToInventory(@NotNull final IInventory inventory, @NotNull final ItemStack itemStack, @NotNull final AbstractBuilding building)
+    public static int getItemCountInProvider(@NotNull final ICapabilityProvider provider, @NotNull final Predicate<ItemStack> itemStackSelectionPredicate)
     {
-        if (!addItemStackToInventory(inventory, itemStack))
-        {
-            final List<ItemStorage> localAlreadyKept = new ArrayList<>();
-            for (int i = 0; i < inventory.getSizeInventory(); i++)
-            {
-                final ItemStack localStack = inventory.getStackInSlot(i);
-                if (!EntityAIWorkDeliveryman.workerRequiresItem(building, localStack, localAlreadyKept))
-                {
-                    final ItemStack removedStack = inventory.removeStackFromSlot(i);
-                    inventory.setInventorySlotContents(i, itemStack.copy());
-                    return removedStack.copy();
-                }
-            }
-        }
-        return null;
+        return getItemHandlersFromProvider(provider).stream()
+                 .mapToInt(handler -> filterItemHandler(handler, itemStackSelectionPredicate).stream().mapToInt(InventoryUtils::getItemStackSize).sum())
+                 .sum();
     }
 
     /**
-     * Adapted from {@link net.minecraft.entity.player.InventoryPlayer#addItemStackToInventory(ItemStack)}.
+     * Checks if a player has a block in the {@link ICapabilityProvider}.
+     * Checked by {@link #getItemCountInProvider(ICapabilityProvider, Block,
+     * int)} &gt; 0;
      *
-     * @param inventory Inventory to add itemstack to.
-     * @param itemStack ItemStack to add.
-     * @return True if successful, otherwise false.
+     * @param Provider   {@link ICapabilityProvider} to scan
+     * @param block      Block to count
+     * @param itemDamage the damage value.
+     * @return True when in {@link ICapabilityProvider}, otherwise false
      */
-    public static boolean addItemStackToInventory(@NotNull final IInventory inventory, @Nullable final ItemStack itemStack)
+    public static boolean hasItemInProvider(@NotNull final ICapabilityProvider Provider, @NotNull final Block block, int itemDamage)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00117)